evap/contributor/tests/test_views.py

```diff
@@ -259,3 +259,11 @@ def test_information_message(self):
             "Please review the evaluation's details below, add all contributors and select suitable questionnaires. "
             "Once everything is okay, please approve the evaluation on the bottom of the page.",
         )
+
+    def test_display_request_buttons(self):
+        self.evaluation.name_en = "Adam & Eve"
+        self.evaluation.allow_editors_to_edit = False
+        self.evaluation.save()
+        page = self.app.get(self.url, user=self.responsible)
+        self.assertIn("Request changes", page)
+        self.assertNotIn("Request creation of new account", page)
```

Comment:
With this (unfinished) test I wanted to make sure that only one button is shown, in this case, that the "Request changes" button is visible while the "Request creation of new account" button is not. The problem is that in this view, the "Request creation of new account" button exists for contributors as well, so the string is always present in the page.
That's why I think I need to parse HTML in order to test whether the button is in the right location. 
Is there a way to do that already? 
If no, can we maybe add a dependency that allows me to parse HTML in tests? @niklasmohrin expressed support for adding beautiful soup, arguing that it would also be useful in other cases.

Reply:
You could assert that "Request creation of new account" is on the page twice when `allow_editors_to_edit` is True and only once when `allow_editors_to_edit` is False. We do something similar in [`test_textanswer_review_state_tags`](https://github.com/e-valuation/EvaP/blob/main/evap/staff/tests/test_views.py#L589)

Reply:
Thank you for the hint!